minibot.py

```diff
@@ -0,0 +1,272 @@
+    single class instance for each joystick and the mapping of ports to
+    hardware buttons depends on the code in the drivestation.
+    """
+
+    def __init__(self, port):
+        """Construct an instance of a joystick.
+        The joystick index is the USB port on the Driver Station.
+
+        :param port: The port on the Driver Station that the joystick is
+            plugged into.
+        :type  port: int
+        """
+        self.port = port
+
+        # Class-instance of driver station
+        if not hasattr(Joystick, 'ds') or Joystick.ds is not None:
+            Joystick.ds = DriverStation()
+
+    def getX(self, hand=None):
+        """Get the X value of the joystick.
+
+        This depends on the mapping of the joystick connected to the current
+        port.
+
+        :param hand: Unused
+        :returns: The X value of the joystick.
+        :rtype: float
+        """
+        return self.getRawAxis(0)
+
+    def getY(self, hand=None):
+        """Get the Y value of the joystick.
+
+        This depends on the mapping of the joystick connected to the current
+        port.
+
+        :param hand: Unused
+        :returns: The Y value of the joystick.
+        :rtype: float
+        """
+        return self.getRawAxis(1)
+
+    def getZ(self, hand=None):
+        return self.getRawAxis(3)
+
+    def getTwist(self):
+        """Get the twist value of the current joystick.
+
+        This depends on the mapping of the joystick connected to the current
+        port.
+
+        :returns: The Twist value of the joystick.
+        :rtype: float
+        """
+        return self.getRawAxis(2)
+
+    def getRawAxis(self, axis):
+        """Get the value of the axis.
+
+        :param axis: The axis to read, starting at 0.
+        :type  axis: int
+        :returns: The value of the axis.
+        :rtype: float
+        """
+        return Joystick.ds.getStickAxis(self.port, axis)
+
+
+class MinibotMotorController(wpilib.PWMSpeedController):
+    """Minibot Speed Controller via PWM"""
+
+    def __init__(self, channel):
+        """Constructor for a minibot electronic speed controller (ESC)
+
+        :param channel: The PWM channel that the ESC is attached to. Channels
+        0-5 are valid channels on the micro maestro.
+        :type  channel: int
+
+        .. note ::
+            These values for the ESC deadbands are untested and likely to need
+             additional calibration
+
+            - 1.800ms = full "forward" - this is a WAG, needs to be found
+             experimentally
+            - 1.595ms = the "high end" of the deadband range
+            - 1.500ms = center of the deadband range (off) - should be roughly
+             correct but could use some fine tuning
+            - 1.460ms = the "low end" of the deadband range
+            - 1.200ms = full "reverse" - also a WAG, pretty much
+             totally random number
+        """
```

Comment:
Do these notes describe the ESCs on the minibot, or are they just copied from somewhere else? If they don't yet describe the minibot ESCs, please note that in the comment.

Reply:
They are for the minibot ESCs, I got the specific values from Ben. He only had values for the upper and lower end of the deadband, not the upper/lower limits on the total range.

Reply:
They are for the minibot ESCs, I got the specific values from Ben. He only had values for the upper and lower end of the deadband, not the upper/lower limits on the total range.